services/ui_backend_service/api/autocomplete.py

```diff
@@ -1,21 +1,43 @@
 from services.utils import handle_exceptions
-from services.data.db_utils import translate_run_key
+from services.data.db_utils import DBResponse, DBPagination, translate_run_key
 from .utils import format_response, format_response_list, web_response, custom_conditions_query, pagination_query
+import sys
+import asyncio
+import re
+
+TAGS_FILL_INTERVAL_SECONDS = 60 * 5
 
 
 class AutoCompleteApi(object):
-    # Cache tags so we don't have to request DB everytime
-    tags = None
 
     def __init__(self, app, db):
         self.db = db
         # Cached resources
+        # Cache tags so we don't have to request DB everytime
+        self.tags = []
         app.router.add_route("GET", "/tags/autocomplete", self.get_tags)
         # Non-cached resources
         app.router.add_route("GET", "/flows/autocomplete", self.get_flows)
         app.router.add_route("GET", "/flows/{flow_id}/runs/autocomplete", self.get_runs_for_flow)
         app.router.add_route("GET", "/flows/{flow_id}/runs/{run_id}/steps/autocomplete", self.get_steps_for_run)
         app.router.add_route("GET", "/flows/{flow_id}/runs/{run_id}/artifacts/autocomplete", self.get_artifacts_for_run)
+        loop = asyncio.get_event_loop()
+        loop.create_task(self.periodic_tags_fetch_and_cache())
+
+    async def periodic_tags_fetch_and_cache(self):
+        '''
+        Async task that fill tags cache every 5minutes. Database query might take a while
+        so its better to cache the result.
+        '''
+        while True:
+            # Get all tags that are mentioned in runs table
+            res, _ = await self.db.run_table_postgres.get_tags()
+            if res.response_code == 200:
+                self.tags = sorted(res.body)
+            size = sys.getsizeof(self.tags) // 1024 // 1024
+            print("SIZE OF TAGS IN MEM: {} Mb".format(size), flush=True)
```

Comment:
Maybe use logger here at debug or info level

Reply:
this is just some leftovers from the WIP. Would it be beneficial to actually log out the object size after fetching though? There's something odd going on with sys.getsizeof as well, seems to work locally, but reports 0 mb on the deployed version. for whatever reason.

Reply:
changed this to use a logger. printing out the memory used, and the number of tags stored with an `INFO` log level